src/CodeGen_LLVM.cpp

```diff
@@ -1804,6 +1804,17 @@ Value *CodeGen_LLVM::codegen_buffer_pointer(Value *base_address, Halide::Type ty
         index = promote_64(index);
     }
 
+    // Peel off a constant offset as a second GEP. This helps LLVM's
+    // aliasing analysis, especially for backends that do address
+    // computation in 32 bits but use 64-bit pointers.
```

Comment:
out of curiosity, what backends are those? Seems like an odd choice.

Reply:
CUDA. pointers are 64-bit but 32-bit integer arithmetic is cheaper than 64-bit